src/log_recv_qtc_to_disk.c

```diff
@@ -0,0 +1,191 @@
+	    for(qpos=0; qpos<100; qpos++) {
+		qtclogline[qpos] = 32;
+	    }
+
+	    qpos = 0;
+	    // QTC:  3799 PH 2003-03-23 0711 YB1AQS        001/10     DL8WPX        0330 DL6RAI        1021
+	    // QTC: 21086 RY 2001-11-10 0759 HA3LI           1/10     YB1AQS        0003 KB3TS          003
+
+	    sprintf(temp, "%3s", band[bandinx]);
+	    if (trxmode == CWMODE) {
+		strcat(temp, "CW  ");
+	    }
+	    else if (trxmode == SSBMODE) {
+		strcat(temp, "SSB ");
+	    }
+	    else {
+		strcat(temp, "DIG ");
+	    }
+	    strncpy(qtclogline, temp, strlen(temp));
+
+	    qpos = strlen(temp);
+	    sprintf(temp, "%04d", qsonr);
+	    strncpy(qtclogline+qpos, temp, strlen(temp));
+	    qpos += strlen(temp);
+
+	    //get_time();
+	    //strftime(time_buf, 60, " %d-%b-%y %H:%M ", time_ptr);
+	    sprintf(time_buf, " %s ", qtcreclist.qtclines[i].receivedtime);
+
+	    //strncpy(qtclogline+qpos, time_buf, strlen(time_buf));
+	    strncpy(qtclogline+qpos, time_buf, strlen(time_buf));
+	    qpos+=strlen(time_buf);
+	    //qpos+=strlen(qtcreclist.qtclines[i].receivedtime);
+
+	    if (lan_active == 1) {
+		qtclogline[qpos++] = thisnode;	// set node ID...
+	    } else {
+		qtclogline[qpos++] = ' ';
+	    }
+	    qtclogline[qpos++] = ' ';
+
+	    sprintf(temp, "%-14s", qtcreclist.callsign);
+
+	    strncpy(qtclogline+qpos, temp, strlen(temp));
+	    qpos+=strlen(temp);
+
+	    sprintf(temp, " %04d", qtcreclist.serial);
+	    strncpy(qtclogline+qpos, temp, strlen(temp));
+	    qpos+=strlen(temp);
+
+	    sprintf(temp, " %04d", qtcreclist.count);
+	    strncpy(qtclogline+qpos, temp, strlen(temp));
+	    qpos+=strlen(temp);
+
+	    sprintf(temp, " %s", qtcreclist.qtclines[i].time);
+	    strncpy(qtclogline+qpos, temp, strlen(temp));
+	    qpos+=strlen(temp);
+
+	    sprintf(temp, " %-14s", qtcreclist.qtclines[i].callsign);
+
+	    strncpy(qtclogline+qpos, temp, strlen(temp));
+	    qpos+=strlen(temp);
+
+	    tempi = atoi(qtcreclist.qtclines[i].serial);
+	    if(tempi < 1000) {
+		sprintf(temp, "  %03d    ", tempi);
+	    }
+	    else {
+		sprintf(temp, " %04d    ", tempi);
+	    }
+	    strncpy(qtclogline+qpos, temp, strlen(temp));
+	    qpos+=strlen(temp);
+	    
+	    if (trx_control == 1) {
+		snprintf(khz, 8, "%7.1f", freq);
+	    }
+	    else {
+		snprintf(khz, 8, "      *");
+	    }
+	    strncpy(qtclogline+qpos, khz, strlen(khz));
+	    qpos += strlen(khz);
+
+	    qtclogline[qpos] = '\0';
+
+	    store_recv_qtc(qtclogline);
+
+	    // send qtc to other nodes......
+	    if (lan_active == 1) {
+	      send_lan_message(QTCRENTRY, qtclogline);
+	    }
+	}
+    }
+    for(i=0; i<10; i++) {
+	qtcreclist.qtclines[i].time[0] = '\0';
+	qtcreclist.qtclines[i].callsign[0] = '\0';
+	qtcreclist.qtclines[i].serial[0] = '\0';
+	qtcreclist.qtclines[i].status = 0;
+	qtcreclist.qtclines[i].confirmed = 0;
+	qtcreclist.qtclines[i].receivedtime[0] = '\0';
+    }
+    for(i=0; i<13; i++) {
+	qtc_ry_lines[i].content[0] = '\0';
```

Comment:
that maybe out of bounds. There are only 12 members of the qtc_ry_lines array defined but 13 (0..12) gets initialized.